components/logins/src/db.rs

```diff
@@ -373,33 +348,93 @@ impl LoginDb {
         let rows_changed = self.execute_named(
             &sql,
             named_params! {
-                ":origin": login.origin,
-                ":http_realm": login.http_realm,
-                ":form_action_origin": login.form_action_origin,
-                ":username_field": login.username_field,
-                ":password_field": login.password_field,
-                ":username_enc": login.username_enc,
-                ":password_enc": login.password_enc,
-                ":guid": login.guid(),
+                ":origin": login.fields.origin,
+                ":http_realm": login.fields.http_realm,
+                ":form_action_origin": login.fields.form_action_origin,
+                ":username_field": login.fields.username_field,
+                ":password_field": login.fields.password_field,
                 ":time_created": login.time_created,
                 ":times_used": login.times_used,
                 ":time_last_used": login.time_last_used,
                 ":time_password_changed": login.time_password_changed,
-                ":local_modified": now_ms,
+                ":local_modified": login.time_created,
+                ":enc_fields": login.enc_fields,
+                ":guid": login.guid(),
             },
         )?;
-        if rows_changed == 0 {
-            log::error!(
-                "Record {:?} already exists (use `update` to update records, not add)",
-                login.guid()
-            );
-            throw!(ErrorKind::DuplicateGuid(login.guid().into_string()));
-        }
+        assert_eq!(
+            rows_changed, 1,
+            "adding a new record must have changed 1 row"
+        );
+        Ok(())
+    }
+
+    fn update_existing_login(&self, login: &Login) -> Result<()> {
+        // assumes the "local overlay" exists, so the guid must too.
+        let sql = format!(
+            "UPDATE loginsL
```

Comment:
ditto here - no "OR" clauses because all callers of this should be 100% confident there's no existing record with the same guid.